x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPlugin.java

```diff
@@ -138,12 +138,28 @@ public class EnrichPlugin extends Plugin implements SystemIndexPlugin, IngestPlu
         Setting.Property.NodeScope
     );
 
+    public static final String CACHE_SIZE_SETTING_BWC_NAME = "enrich.cache.size";
+    public static final Setting<FlatNumberOrByteSizeValue> CACHE_SIZE_BWC = new Setting<>(
+        CACHE_SIZE_SETTING_BWC_NAME,
+        (String) null,
+        (String s) -> FlatNumberOrByteSizeValue.parse(
+            s,
+            CACHE_SIZE_SETTING_BWC_NAME,
+            new FlatNumberOrByteSizeValue(ByteSizeValue.ofBytes((long) (0.01 * JvmInfo.jvmInfo().getConfiguredMaxHeapSize())))
+        ),
+        Setting.Property.NodeScope,
+        Setting.Property.Deprecated
+    );
+
     private final Settings settings;
     private final EnrichCache enrichCache;
 
     public EnrichPlugin(final Settings settings) {
         this.settings = settings;
         FlatNumberOrByteSizeValue maxSize = CACHE_SIZE.get(settings);
+        if (maxSize == null) {
```

Comment:
I think this deserves a test now so that show we can work with either or both settings used

Reply:
@dakrone I remember we talked about adding a test when this was up for review, but we figured it was pretty awkward to assert this properly without too many shenanigans. Do you have any other/new thoughts on this? I still feel like storing the `maxSize` as a field somewhere just so we can assert it in a test, feels a little off.

Reply:
In this case I think exposing the `maxSize` to ensure that it has fallen back in the correct manner makes sense.